soc/arm/silabs_exx32/common/soc.c

```diff
@@ -189,6 +200,18 @@ static int silabs_exx32_init(const struct device *arg)
 	/* disable interrupts */
 	oldLevel = irq_lock();
 
+#ifdef CONFIG_SOC_GECKO_DEV_INIT
+	CHIP_Init();
+	sl_board_preinit();
+	sl_device_init_dcdc();
+	sl_device_init_hfxo();
+	sl_device_init_lfxo();
+	sl_device_init_dpll();
+	sl_device_init_clocks();
+	sl_device_init_emu();
+	sl_board_init();
```

Comment:
I've just checked what sl_board_init() does, and we can't definitely accept something like this in Zephyr. https://github.com/zephyrproject-rtos/hal_silabs/pull/19/files#diff-732e868df2ebaa0cb659a0f52f366c33d4b9f9bbb44416e351732fc24df87c40R45

Zephyr already has a mechanism to do these things (boards), and this clearly deviates from what IMO is acceptable. How is going to be generic for all Zephyr boards if put under soc?

Unfortunately, we have historically pulled HAL code without control/audit to make "progress", and this slows progress on core features later (e.g. pinctrl, clock control, device model, etc.).

Reply:
@gmarull from what I know, the purpose of the whole init process enclosed under `#ifdef CONFIG_SOC_GECKO_DEV_INIT` is meant to represent the init process for the board as it is done in Simplicity Studio. I understand that it is not the way it should be done in Zephyr. Initially it was done this way to speed up the develepment. Would it be ok for now if I move this sequence to board-specific init?

Reply:
`sl_board_preinit` was completely removed. Instead of this I placed GPIO clock enable in the common initialization part of the GPIO driver. I also had to set the GPIO driver initialization stage to `EARLY` - UART driver depends on correct GPIO clock configuration and UART driver is initialized at `PRE_KERNEL_1`.
`sl_board_init` and `sl_device_init_clocks` were moved to board init function in `board.c` because those functions are board specific.